File: arrays_and_hashing/two_sum.py
# Memory: 7.7 MB•Time: 27ms•Submitted at: 04/15/2026 14:14
# Beats 100% in runtime, 99.97% in memory

# Given an array of integers nums and an integer target, return the indices i and j such that nums[i] + nums[j] == target and i != j.

# You may assume that every input has exactly one pair of indices i and j that satisfy the condition.

# Return the answer with the smaller index first. 

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []

        for i, first in enumerate(nums):
            j = i + 1
            while j < len(nums):
                if first + nums[j] == target:
                    result.append(i)
                    result.append(j)
                    return result
                j += 1

        return result;
                    

File: arrays_and_hashing/two_sum.cpp
// Memory: 105.9 MB•Time: 2ms•Submitted at: 04/15/2026 13:25
// Beats 99.27% in run time, 97.36% in memory

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        vector<int> results;

        for (int i = 0; i < nums.size(); i++) {
            for (int j = i+1; j <= nums.size(); j ++) {
                if (nums[i] + nums[j] == target) {
                    results.push_back(i);
                    results.push_back(j);
                    return results; 
                }
            }
        }

        return results;

    }
};
